platform/msm_shared/dev_tree.c

```diff
@@ -561,7 +567,23 @@ void *dev_tree_appended(void *kernel, uint32_t kernel_size, uint32_t dtb_offset,
 	}
 
 	if(bestmatch_tag) {
-		memcpy(tags, bestmatch_tag, bestmatch_tag_size);
+#if WITH_LK2ND
+		/* If the DTB is a container, unpack it. */
+		/* Most trees < 64k so 128k should be enough. */
+		memcpy(tags + 1024*128, bestmatch_tag, bestmatch_tag_size);
```

Comment:
Instead of using semi-magical numbers (1024, 128) add some human-readable constant/macro and use it in all places